Graphs/LeafletGraph/index.js

```diff
@@ -0,0 +1,339 @@
+          const source = currentMarkers.find(marker => marker.name === line[links.sourceColumn]);
+          if (!destMarker && dest) {
+            destMarker = dest.position;
+          }
+
+          if (!sourceMarker && source) {
+            sourceMarker = source.position;
+          }
+
+          // if any source or destintion point not found in bound, then find lat lng in data
+          if (!destMarker && sourceMarker) {
+            
+            const destination = stateData.find(d => d[idColumn] === line[links.destinationColumn])
+
+            if (destination) {
+              destMarker = {
+                lat: destination[latitudeColumn],
+                lng: destination[longitudeColumn],
+              }
+            }
+          }
+
+          // if any source or destintion point not found in bound, then find lat lng in data
+          if (!sourceMarker && destMarker) {
+            const source = stateData.find(d => d[idColumn] === line[links.sourceColumn]);
+
+            if (source) {
+              sourceMarker = {
+                lat: source[latitudeColumn],
+                lng: source[longitudeColumn],
+              }
+            }
+          }
+
+          if (destMarker && sourceMarker) {
+            if (!isPolylineExist(polylines, sourceMarker, destMarker)) {
+              polylines.push({
+                source: { lat: sourceMarker.lat, lng: sourceMarker.lng },
+                destination: { lat: destMarker.lat, lng: destMarker.lng },
+                color: line.color || theme.palette.redColor,
+              })
+            }
+          }
+        }
+
+      })
+      setLine(polylines);
+    }
+  }
+
+  // check line is already drawn or not
+  const isPolylineExist = (polylines, sourceMarker, destMarker) => {
+    return polylines.some(el => {
+      return ((el.source.lat === sourceMarker.lat && el.source.lng === sourceMarker.lng) && (el.destination.lat === destMarker.lat && el.destination.lng === destMarker.lng));
+    });
+  }
+
+  const renderPolylineIfNeeded = () => {
+    return line.map((link, i) => {
+      return <Polyline
+        color= {link.color}
+        positions={[
+          [link.source.lat, link.source.lng],
+          [link.destination.lat, link.destination.lng]
+        ]}
+      />
+    })
+  }
+
+  const createClusterCustomIcon = (clusters) => {
+    const count = clusters.getChildCount();
+
+    return L.divIcon({
+      html:
+        `<div>
+              <img src='/icons/GREY.png'>
+                <span class="markerClusterLabel">${count}</span>
+              </img>
+          </div>`,
+      className: 'markerCluster',
+    });
+  }
+
+  const currentCenter = [
+    Number(process.env.REACT_APP_MAP_LAT),
+    Number(process.env.REACT_APP_MAP_LNG),
+  ]
+
+  if (searchBar !== false) {
+    height -= 70;
+  }
+
+  return (
+    <React.Fragment>
+      {renderSearchBarIfNeeded()}
```

Comment:
Minor: Can you rename these functions?
renderSearchBarIfNeeded -> renderSearchBar
renderMarkersIfNeeded -> renderMarkers
renderPolylineIfNeeded -> renderPolyline